torch/csrc/jit/codegen/cuda/parser.cpp

```diff
@@ -613,116 +616,359 @@ class IrParser {
           });
     }
 
+    {
+      std::array<const char*, kNumBatchnormFwd> BatchNormFwd = {
+          "aten::_batch_norm_impl_index(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, bool cudnn_enabled) -> (Tensor, Tensor, Tensor, Tensor, int)",
+          "aten::native_batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps) -> (Tensor, Tensor, Tensor)",
+          "aten::batch_norm(Tensor input, Tensor? weight, Tensor? bias, Tensor? running_mean, Tensor? running_var, bool training, float momentum, float eps, bool cudnn_enabled) -> Tensor"};
+      for (auto signature : BatchNormFwd) {
+        auto ptr_op = getOperatorForLiteral(signature);
+        registerParseRule(
+            ptr_op,
+            [](const Node* node,
+               std::unordered_map<size_t, CgValue>& value_map) -> void {
+              auto input =
+                  value_map[node->input(0)->unique()]->as<TensorView>();
+
+              // TODO: it feels quite sketchy to modify fusion from parser
+              auto fusion = FusionGuard::getCurFusion();
```

Comment:
It actually seems okay in this instance, but you can grab the fusion from any value simply with `val->fusion()`. In order to do all the parsing below though you need the fusion active anyways.